toolchain/lexer/tokenized_buffer.h

```diff
@@ -253,7 +253,7 @@ class TokenizedBuffer {
   class TokenLocationTranslator
       : public DiagnosticLocationTranslator<Internal::TokenizedBufferToken> {
    public:
-    explicit TokenLocationTranslator(TokenizedBuffer& buffer,
+    explicit TokenLocationTranslator(const TokenizedBuffer& buffer,
```

Comment:
Sadly, we need to use pointers here (and below) I think, because the parameter has to outlive the constructor call, and when its `const`, that's not guaranteed...

```suggestion
    explicit TokenLocationTranslator(const TokenizedBuffer* buffer,
```